### PR TITLE
feat: SplitJoinOrigin

### PR DESCRIPTION
**Pull request by whyy7777**
1, OriginSplitJoin
2, EmitAllWindow

### CODE REVIEW

**Review by wzru — CHANGES_REQUESTED** [review]
Use clang-format to format codes :)

File: include/Common/Result.hpp [hunk 1]
```diff
@@ -41,7 +41,7 @@ struct JoinResult
     std::vector<std::mutex> mu;
     JoinResult(const Param &param);
     void Emit(int wid, TuplePtr t1, TuplePtr t2);
-    void Emit(TuplePtr t1, TuplePtr t2);
+    void EmitAllWindow(TuplePtr t1, TuplePtr t2);
```
[the remark this answers is not in the document]
Comment (whyy7777) [reply include/Common/Result.hpp:44]: I think it is more explicit after renaming

File: .gitignore [hunk 2]
```diff
@@ -34,3 +34,4 @@ com_scripts
 commit
 .idea/
 .vscode/
+./DS_Store
```
[the remark this answers is not in the document]
Comment (whyy7777) [reply .gitignore]: done

File: .gitignore [hunk 3]
```diff
@@ -34,3 +34,4 @@ com_scripts
 commit
 .idea/
 .vscode/
+.DS_Store/
```
Comment (wzru) [inline .gitignore:37]: .DS_Store still exists...

File: test/System/Test.cpp [hunk 4]
```diff
@@ -74,6 +74,27 @@ TEST(SystemTest, SplitJoin)
     S->Load();
     auto engine = make_unique<EagerEngine>(param);
     engine->Run(ctx);
+    ctx.res->Print();
```
Comment (wzru) [inline test/System/Test.cpp:77]: remove this

Comment (whyy7777) [reply test/System/Test.cpp]: done

File: .gitignore (hunk 3 above)
Comment (whyy7777) [reply .gitignore]: done

**Review by wzru — APPROVED** [review]

**Pull request merged**